pkg/compact/compact.go

```diff
@@ -1029,19 +1028,18 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 
 		func(ctx context.Context, meta *metadata.Meta) {
 			g.Go(func() error {
-				tracing.DoInSpanWithErr(ctx, "compaction_block_download", func(ctx context.Context) error {
-					err = block.Download(ctx, cg.logger, cg.bkt, meta.ULID, bdir, objstore.WithFetchConcurrency(cg.blockFilesConcurrency))
-					return err
+				err := tracing.DoInSpanWithErr(ctx, "compaction_block_download", func(ctx context.Context) error {
+					return block.Download(ctx, cg.logger, cg.bkt, meta.ULID, bdir, objstore.WithFetchConcurrency(cg.blockFilesConcurrency))
 				}, opentracing.Tags{"block.id": meta.ULID})
 				if err != nil {
 					return retry(errors.Wrapf(err, "download block %s", meta.ULID))
 				}
 
 				// Ensure all input blocks are valid.
 				var stats block.HealthStats
-				tracing.DoInSpanWithErr(ctx, "compaction_block_health_stats", func(ctx context.Context) error {
-					stats, err = block.GatherIndexHealthStats(cg.logger, filepath.Join(bdir, block.IndexFilename), meta.MinTime, meta.MaxTime)
-					return err
+				err = tracing.DoInSpanWithErr(ctx, "compaction_block_health_stats", func(ctx context.Context) (e error) {
+					stats, e = block.GatherIndexHealthStats(cg.logger, filepath.Join(bdir, block.IndexFilename), meta.MinTime, meta.MaxTime)
+					return e
```

Comment:
ditto

Reply:
I cannot do this here right, as im assigning stats as well (to make it worst stats is defined outside the function so i cannot use `:=`)

What i can do here (but idk if you prefer) is:

```
				if err := tracing.DoInSpanWithErr(ctx, "compaction_block_health_stats", func(ctx context.Context) (e error) {
					stats, e = block.GatherIndexHealthStats(cg.logger, filepath.Join(bdir, block.IndexFilename), meta.MinTime, meta.MaxTime)
					return e
				}, opentracing.Tags{"block.id": meta.ULID}); err != nil {
					return errors.Wrapf(err, "gather index issues for block %s", bdir)
				}
```